binary_search_tree/tree.py

```diff
@@ -8,52 +8,116 @@ def __init__(self, key, val = None):
         self.left = None
         self.right = None
         
-
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: worst case O(N)
+    # Space Complexity: O(1)
     def add(self, key, value = None):
```

Comment:
✨ Nice iterative solution! Yes, time complexity would be O(n) if the tree is unbalanced, O(log n) if balanced.